core/src/test/java/org/bitcoinj/crypto/HDPathTest.java

```diff
@@ -203,8 +203,13 @@ private PathVector[] parseTestVectors() {
     @Test
     @Parameters(method = "toStringTestVectors, parseTestVectors")
     public void testParsePath(PathVector tv) {
-        HDPath.HDFullPath path = HDPath.parsePath(tv.pathString);
+        HDPath path = HDPath.parsePath(tv.pathString);
         assertEquals(tv.path, path);
+        // TODO: This is a reminder to implement equals and hashCode on HDPath and subclasses
```

Comment:
Can we merge equals/hashCode first, then?

Reply:
Yes, we did.

Reply:
Not the full version.

Reply:
I think we have to implement (almost) all of the change in PR #3721 before we can start comparing `hasPrivateKey` in `equals`/`hashCode` -- the code and tests are full of assumptions left over from when we were using `List<ChildNumber>` everywhere. We have to declare stricter types and make sure we convert between them before doing a full comparison.

Reply:
I'm also happy to remove the comment and "kludge" if you will merge this PR without them. We don't really need the reminder we have pending PRs with the fix.